tests/debugger/test_debugger_probe_snapshot.py

```diff
@@ -132,3 +140,57 @@ def test_code_origin_entry_present(self):
                 code_origins_entry_found = code_origin_type == "entry"
 
         assert code_origins_entry_found
+
+    def setup_log_line_probe_snaphots_budgets(self):
+        self._setup(
+            "probe_snapshot_log_line_budgets",
+            "/debugger/budgets/20",
+            lines=self.method_and_language_to_line_number("Budgets", self.get_tracer()["language"]),
+        )
+
+    @missing_feature(context.library == "dotnet", reason="Probe snapshot budgets are not yet implemented")
+    @missing_feature(context.library == "nodejs", reason="Probe snapshot budgets are not yet implemented")
+    @missing_feature(context.library == "ruby", reason="Probe snapshot budgets are not yet implemented")
+    def test_log_line_probe_snaphots_budgets(self):
+        self._assert()
+        self._validate_snapshots()
+
+        snapshots = 0
+        for _id in self.probe_ids:
+            for span in self.probe_snapshots[_id]:
+                snapshot = span.get("debugger", {}).get("snapshot", None)
+                if snapshot is None:
+                    continue
+
+                snapshots += 1
+
+            # Line probes with snapshots should output 1 log per second. For a loop of 20 iterations, we should have
+            # only have enriched log with a snapshot.
+            assert snapshots == 1, f"Expected 1 snapshots, got {snapshots}"
```

Comment:
@jpbempel @ P403n1x87 Is there something incorrect in my assumption here? I thought for snapshot probes, we rate-limit to 1/s? I'm seeing ~16 snapshots from both the Java and Python tracers